src/fourcipp/fourc_input.py

```diff
@@ -65,6 +66,42 @@ def is_section_known(section_name: str) -> bool:
 CONVERTER = Converter()
 
 
+def modify_input_with_defaults(input_path: pathlib.Path, overwrite_flag: bool) -> None:
+    """Apply user defaults to an input file located at input_path.
+
+    Args:
+         input_path: Input filename to apply user defaults to.
+         overwrite_flag: Whether to overwrite the existing input file.
+                         By default, a new file with suffix '_mod.4C.yaml' is created.
+    """
+    OUTPUT_SUFFIX = "_mod.4C.yaml"
+
+    if not input_path.is_file():
+        raise FileNotFoundError(f"Input file '{input_path}' does not exist.")
+    input_data = FourCInput.from_4C_yaml(input_path)
+    user_defaults_string = CONFIG["user_defaults_path"]
+    if user_defaults_string is None:
+        raise ValueError("User defaults path is not set in the config.")
+    user_defaults_path = pathlib.Path(user_defaults_string)
+    if not user_defaults_path.is_file():
+        raise FileNotFoundError(
+            f"User defaults file '{user_defaults_path}' does not exist."
+        )
+    logger.info(f"Applying user defaults from '{user_defaults_path}''.")
+    input_data.apply_user_defaults(user_defaults_path)
+    if overwrite_flag:
+        output_filename = input_path
+    else:
+        input_string = str(input_path)
+        if input_string.endswith(".4C.yaml"):
+            output_filename = pathlib.Path(input_string[:-8] + OUTPUT_SUFFIX)
+        else:
+            logger.warning(f"Input file '{input_string}' does not end with '.4C.yaml'.")
+            output_filename = pathlib.Path(input_path.stem + OUTPUT_SUFFIX)
```

Comment:
Let's generalize this to other suffixes. Something like this should work independently of the suffixes. Using `4C.yaml` isn't mandatory, and if you have a partial file, i.e., that is used within an include, you might have additional suffixes.

```suggestion
        names = input_path.name.split(".")
        names = names[0]+ output_prefix + names[1:]
        output_filename = input_path.with_name(".".join(names))
```

Reply:
ups I just realised, this should be:
```python
names=[names[0]+ output_prefix] + names[1:]
```